lib/ngtsc/workspace.symbols.ts

```diff
@@ -0,0 +1,516 @@
+        this.options.preserveWhitespaces || false,
+        this.options.i18nUseExternalIds !== false,
+        this.options.enableI18nLegacyMessageIdFormat !== false,
+        this.options.i18nNormalizeLineEndingsInICUs,
+        this.moduleResolver,
+        this.cycleAnalyzer,
+        this.refEmitter,
+        this.defaultImportTracker,
+        this.incrementalDriver.depGraph,
+        this.injectableRegistry,
+        !!this.options.annotateForClosureCompiler,
+      )
+    );
+  }
+
+  /** Static reflection of declarations using the TypeScript type checker */
+  public get reflector() {
+    return this.lazy('reflector', () => new TypeScriptReflectionHost(this.checker));
+  }
+
+  /** Evaluate typecript Expression & update the dependancy graph accordingly */
+  public get evaluator() {
+    return this.lazy('evaluator', () => new PartialEvaluator(
+      this.reflector,
+      this.checker,
+      this.incrementalDriver.depGraph
+    ));
+  }
+
+  /** Register metadata from local NgModules, Directives, Components, and Pipes */
+  public get metaRegistry() {
+    return this.lazy('metaRegistry', () => new CompoundMetadataRegistry([ this.localMetaReader, this.scopeRegistry ]));
+  }
+
+  /** Register metadata from local declaration files (.d.ts) */
+  public get metaReader() {
+    return this.lazy('metaReader', () => new CompoundMetadataReader([ this.localMetaReader, this.dtsReader ]));
+  }
+
+  /** Collects information about local NgModules, Directives, Components, and Pipes (declare in the ts.Program) */
+  public get scopeRegistry() {
+    return this.lazy('scopeRegistry', () => {
+      const depScopeReader = new MetadataDtsModuleScopeResolver(this.dtsReader, this.aliasingHost);
+      return new LocalModuleScopeRegistry(this.localMetaReader, depScopeReader, this.refEmitter, this.aliasingHost);
+    });
+  }
+
+  /** Analyze lazy loaded routes */
+  public get routeAnalyzer() {
+    return this.lazy('routeAnalyzer', () => new NgModuleRouteAnalyzer(this.moduleResolver, this.evaluator));
+  }
+
+  /** Find a symbol based on the class expression */
+  public findSymbol(token: Expression) {
+    return findSymbol(this, token);
+  }
+  /** Find a symbol based on the class expression */
+  public getSymbol(node: ClassDeclaration) {
+    return getSymbolOf(this, node);
+  }
+
+  public getClassRecords() {
+    this.ensureAnalysis();
+    return this.traitCompiler.allRecords();
+  }
+
+  public getAllModules() {
+    this.ensureAnalysis();
+    return this.traitCompiler.allRecords('NgModule').map(({ node }) => new NgModuleSymbol(this, node));
+  }
+
+  public getAllComponents() {
+    this.ensureAnalysis();
+    return this.traitCompiler.allRecords('Component').map(({ node }) => new ComponentSymbol(this, node));
+  }
+
+
+  public getAllDirectives() {
+    this.ensureAnalysis();
+    return this.traitCompiler.allRecords('Directive').map(({ node }) => new DirectiveSymbol(this, node));
+  }
+
+  public getAllInjectable() {
```

Comment:
I'd say this is excellent. Having the dependencies of an `Injectable` is enough to cover most use cases, we don't need to dig into the privoders.